src/YamlFrontMatter.php

```diff
@@ -23,6 +23,79 @@ public static function parse(string $content): Document
         return new Document($matter, $body);
     }
 
+    /**
+     * A parser that can handle Markdown that contains Markdown.
+     *
+     * Attempts to follow the practices defined in https://jekyllrb.com/docs/front-matter/.
+     *
+     * Fixes https://github.com/spatie/yaml-front-matter/discussions/30.
+     *
+     * @param string $content
+     * @return Document
+     */
+    public static function markdownCompatibleParse(string $content): Document
```

Comment:
This whole function is kinda long, could you split it up in smaller logical parts?

Reply:
I'll refactor it! I think I will separate it all into its own class as I think it will be easier that way.

*This comment is in reply to all three notations

Reply:
Created a new class, [`ComplexMarkdownParser`](https://github.com/spatie/yaml-front-matter/blob/2176038f1dadefc219d4fc29eca034024f222b3f/src/ComplexMarkdownParser.php) as I felt it would be easier (and make more sense) to work in an object oriented matter. I kept the static `markdownCompatibleParse` function but changed it to offload the logic into the new class.

```php
public static function markdownCompatibleParse(string $content): Document
{
    return (new ComplexMarkdownParser($content))->parse();
}
```

The class still has 100% test coverage (according to PHPStorm) and I also continuously ran smoke tests to ensure that the actual output did not change.